java/pom.xml

```diff
@@ -1058,7 +1058,7 @@
                     -DCMAKE_BUILD_TYPE=Release
                     -DCMAKE_INSTALL_LIBDIR=lib
                     -DCMAKE_INSTALL_PREFIX=${arrow.dataset.jni.dist.dir}
-                    -DCMAKE_PREFIX_PATH=${project.basedir}/../java-dist
+                    -DCMAKE_PREFIX_PATH=${project.basedir}/../${os.detected.arch}/java-dist
```

Comment:
@davisusanibar is this enough to pull the correct modules or do we need to adjust more paths?

Reply:
Hi Rok, I could suggest that is needed to add `${os.detected.arch}` before all places working with folder `java-dist`

The you could test your changes locally with this maven command please: `mvn generate-resources -Pgenerate-libs-jni-macos-linux -N` inside of `arrow/java` folder ?